libs/SalesforceSDK/src/com/salesforce/androidsdk/util/test/JSONTestHelper.java

```diff
@@ -43,28 +43,34 @@
 	 * @throws JSONException
 	 */
 	public static void assertSameJSON(String message, Object expected, Object actual) throws JSONException {
```

Comment:
Refactored that class so that when there is a mismatch, you see the full expected value and the full actual value.